### DISCUSSION TITLE
Delta jobs failing

### DISCUSSION THREAD

**Discussion by bbkofman** [opengisch/QField#3348] [Q&A]
Just starting to use the Cloud product. After updating a point's attribute value in the Android app, I pushed the "Publish changes" button. On the server I see a job of type "Delta Apply". The status goes from Pending to Failed. Tried it a few times with the same result. The log has this:

Start QGIS Application
18:32:07.148 QGSSTDERR INFO     Starting QGIS app version 32601 (exported)...
18:32:07.154 QGSSTDERR WARNING  QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
Download Project Directory
18:32:10.127 root      INFO     Downloading file "Parcels_MASTER-wgs84.dbf"…
18:32:11.096 root      INFO     Downloading file "Parcels_MASTER-wgs84.shp"…
18:32:11.835 root      INFO     Downloading file "Parcels_MASTER-wgs84.shx"…
18:32:12.311 root      INFO     Downloading file "PastTrees.dbf"…
18:32:13.000 root      INFO     Downloading file "PastTrees.shp"…
18:32:13.479 root      INFO     Downloading file "PastTrees.shx"…
18:32:14.268 root      INFO     Downloading file "Roadsdraft-1-wgs84.dbf"…
18:32:14.802 root      INFO     Downloading file "Roadsdraft-1-wgs84.shp"…
18:32:15.322 root      INFO     Downloading file "Roadsdraft-1-wgs84.shx"…
18:32:16.090 root      INFO     Downloading file "Trees-wgs84 2021.kmz"…
18:32:16.618 root      INFO     Downloading file "Trees-wgs84.dbf"…
18:32:17.173 root      INFO     Downloading file "Trees-wgs84.shp"…
18:32:17.734 root      INFO     Downloading file "Trees-wgs84.shx"…
18:32:18.248 root      INFO     Downloading file "Trees.qgz"…
18:32:18.746 root      INFO     Downloading file "tree zones.dbf"…
18:32:19.249 root      INFO     Downloading file "tree zones.shp"…
18:32:20.089 root      INFO     Downloading file "tree zones.shx"…
18:32:20.121 root      INFO     Local files list for project "e7b4985b-afde-4e9f-8ac2-c8ca0e7be0d5":
Name                         Size  MD5 Checksum
------------------------  -------  --------------------------------
Parcels_MASTER-wgs84.dbf  3460550  9362ad2e371f65cae5ddd34ba3ec48e2
Parcels_MASTER-wgs84.shp  1031080  1db7e5f65ee39319ff4b18f2e28c4596
Parcels_MASTER-wgs84.shx    36316  1ce48886ca9cc6965b58aa0fc185c1f8
PastTrees.dbf              199039  200223a9ddd158a983fd191b6f08cb2e
PastTrees.shp               18664  ee2f040249a906b046c4a560eb0f4537
PastTrees.shx                5404  394a9af54214febe273e354f09d7e34b
Roadsdraft-1-wgs84.dbf     168109  91ab5ff2a95587cc9cbf3bddae6740cd
Roadsdraft-1-wgs84.shp     171340  ac5d1aa8fbe323182f2de0f707a11118
Roadsdraft-1-wgs84.shx       4124  1290bb3e0c2c33c5b949c4529daaf8e4
Trees-wgs84 2021.kmz        77711  c0f77c67180283bfe8891f8ac24c295e
Trees-wgs84.dbf            803318  fcd23cced4d810461659311a4bca56a2
Trees-wgs84.shp             75252  4b0bcac285f3dc0dca86e1dc3ba8ef20
Trees-wgs84.shx             21572  6125fa9129608bf16f1e3ec9a3c97de0
Trees.qgz                   46328  c7d7099324742c7b2b5c1bd5c475385e
tree zones.dbf                120  54e0f18fefe5e368b8e2590d0caa262d
tree zones.shp                924  198aeca799c2426bbcfc30da9a7a0ebd
tree zones.shx                140  7a09683bb44d72acb9e8a3017b4bc487
Apply Deltas
18:32:20.122 root      INFO     Loading project file "/tmp/tmpp7dzy162/files/Trees.qgz"...
proj_create: Error -38: failed to load datum shift file
proj_create: Error -38: failed to load datum shift file
18:32:20.784 root      INFO     Loading delta file "/io/deltafile.json"...
18:32:20.809 qfieldcloud.qgis.apply_deltas ERROR    An unknown error has been encountered while applying delta: 
18:32:20.814 QGSSTDERR WARNING QFSFileEngine::open: No file name specified
18:32:20.814 QGSSTDERR INFO Stopping QGIS app…

**Comment by suricactus** [comment]
If you are using experimental data sources, Shapefiles in your case, please make sure you follow the requirements as described here: https://docs.qfield.org/get-started/tutorials/get-started-qfc/?h=shapefile#configure-your-project-layers-for-qfield . Even better - just use geopackages.

**Comment by bbkofman** [comment]
Thank you for the pointers!
Boris
